sssd_test_framework/roles/keycloak.py

```diff
@@ -126,12 +129,66 @@ def add(
 
         return self
 
-    def delete(self) -> None:
+    def delete(
+        self,
+    ) -> SSHProcessResult:
         """
         Delete Keycloak user.
+
+        :return: SSH exec result
+        :rtype: SSHProcessResult
         """
         del_user = f"delete users/{self.id}"
-        self.role.kcadm(del_user)
+        result = self.role.kcadm(del_user)
+        return result
+
+    def modify(
+        self,
+        *,
+        firstName: str | None = None,
+        lastName: str | None = None,
+        email: str | None = None,
+        enabled: bool | None = None,
+    ) -> KeycloakUser:
+        """
+        Modify keycloak user attributes.
+
+        Parameters that are not set are ignored.
+
+        :param firstName: User's first name
+        :type firstName: str | None, optional
+        :param lastName: User's last name
+        :type lastName: str | None, optional
+        :param email: User's email address
+        :type email: str | None, optional
+        :param enabled: User account enabled/disabled status
+        :type enabled: bool | None, optional
+        :return: SSH exec result
+        :rtype: SSHProcessResult
+        """
+
+        attrs = {"firstName": firstName, "lastName": lastName, "email": email, "enabled": enabled}
+
+        update_attrs = " ".join([f'-s "{k}={v}"' for k, v in attrs.items() if v is not None])
+        update_user = f"update users/{self.id} {update_attrs}"
+        self.role.kcadm(update_user)
+        return self
+
+    def set_password(
+        self,
+        password: str | None = "Secret123",
+    ) -> KeycloakUser:
+        """
+        Set Keycloak user password.
+
+        :param password: Password, defaults to 'Secret123'
+        :type password: str | None, optional
+        :return: Self,
+        :rtype: KeycloakUser
+        """
+        set_password = "set-password -r master " f"--username {self.name} " f"--new-password {password}"
```

Comment:
f"set-password -r master --username {self.name} --new-password {password}" 

instead?

Reply:
yes, I think this must have be a reformat mix-up on my part.  I believe I originally had this spread across multiple lines and one of the linter/format tools combined on one line.  I changed it to reflect your suggestion.